crates/turborepo-paths/src/anchored_unix_path_buf.rs

```diff
@@ -0,0 +1,40 @@
+use bstr::{BString, ByteSlice};
+
+use crate::{PathError, RelativeUnixPathBuf};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+pub struct AnchoredUnixPathBuf(BString);
+
+impl AnchoredUnixPathBuf {
+    pub fn into_inner(self) -> BString {
+        self.0
+    }
+
+    pub fn make_canonical_for_tar(&mut self, is_dir: bool) {
+        if is_dir {
+            if !self.0.ends_with(b"/") {
+                self.0.push(b'/');
+            }
+        }
+    }
```

Comment:
After this happens it's not _really_ an `AnchoredUnixPathBuf`. It's an `AnchoredUnixTarPathBuf`. I'd prefer creating a new object here.